packages/@ember/-internals/glimmer/lib/glimmer-component-docs.ts

```diff
@@ -0,0 +1,165 @@
+/**
+  A component is a reusable UI element that consists of a `.hbs` template and an
+  optional JavaScript class that defines its behavior. For example, someone
+  might make a `button` in the template and handle the click behavior in the
+  JavaScript file that shares the same name as the template.
+
+  The APIs available in a component vary depending on whether they import from
+  `@glimmer/component` or the older "classic" type, `@ember/component`. The
+  documentation below covers 100% of the available methods, hooks, and
+  properties of `@glimmer/component`.
+
+  ## Defining a component
+
+  To define a component, subclass `Component` and add your own properties,
+  methods and lifecycle hooks:
+
+  ```javascript
+  import Component from '@glimmer/component';
+
+  export default class SomeComponent extends Component {
+    // your code here
+  }
+  ```
+
+  @module @glimmer/component
+  @public
+*/
+
+/**
+  @class GlimmerComponent
+  @public
+  @static
+*/
+
+/**
+  Constructs a new component and assigns itself the passed properties. The
+  constructor is run whenever a new instance of the component is created, and
+  can be used to setup the initial state of the component.
+
+  ```javascript
+  import Component from '@glimmer/component';
+
+  export default class SomeComponent extends Component {
+    constructor(owner, args) {
+      super(owner, args);
+
+      if (this.args.displayMode === 'list') {
+        this.items = [];
+      }
+    }
+  }
+  ```
+
+  Service injections and arguments are available in the constructor.
+
+  ```javascript
+  import Component from '@glimmer/component';
+  import { inject as service } from '@ember/service';
+
+  export default class SomeComponent extends Component {
+    @service animation;
```

Comment:
```suggestion
    @service myAnimations;
```

I have found it useful to preface things that aren't built into Ember with `my` or `custom` in order to help new people who don't know where the boundaries are.